### PR TITLE
High Level Rest Client support for SLM

### PR DESCRIPTION
**Pull request by dakrone**
This commit add HLRC support for SLM.

Relates to #38461

### CODE REVIEW

**Comment by dakrone** [comment]
This is marked as a draft because it needs to be finished in terms of tests and documentation. I'm opening it in case anyone wants to review while I'm out (or wants to push to it) and to make sure others aren't blocked on it as work.

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/1

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleClient.java [hunk 1]
```diff
@@ -300,4 +306,120 @@ public void retryLifecyclePolicyAsync(RetryLifecyclePolicyRequest request, Reque
         restHighLevelClient.performRequestAsyncAndParseEntity(request, IndexLifecycleRequestConverters::retryLifecycle, options,
             AcknowledgedResponse::fromXContent, listener, emptySet());
     }
+
+    /**
+     * Retrieve one or more snapshot lifecycle policy definitions. See
+     * <a href="fill-me-in">
```
Comment (gwbrown) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleClient.java]: I think we can fill in these links now that the asciidoc is written. It won't be on the website obviously, but it's pretty straightforward to determine the URLs.

I'm okay with leaving this if there's something I'm missing, though.

**Review by AthenaEryma — APPROVED** [review]
LGTM with one note about the links. If you fill in the links, no need for another round of review, if we're not going to fill them in now, same deal.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleClient.java (hunk 1 above)
Comment (dakrone) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleClient.java]: Thanks Gordon, I filled in the links.

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**